Modules/Shop/Http/Services/ShopService.php

```diff
@@ -45,31 +45,28 @@ public function getBrandProducts($request): array
         if ($brandDetails) {
             $userId = $brandDetails->user_id;
             //all products count
-            $allProductsCount = Product::where('user_id', $userId)->where('status', 'publish')->count();
-
+            $allProductsCount = User::find($userId)->getAllProducts()->where('status', 'publish')->count();
             //new products count
-            $newProductsCount = Product::where('user_id', $userId)->where('status', 'publish')->where('created_at', '>', now()->subDays(7)->endOfDay())->count();
-
-            $productCategories = Product::select(DB::raw("count(*) as prdct_count"), "category")->where('status', 'publish')->where('user_id', $userId)->groupBy('category')->get();
+            $newProductsCount = User::find($userId)->getAllProducts()->where('status', 'publish')->where('created_at', '>', now()->subDays(7)->endOfDay())->count();
 
+            $productCategories = User::find($userId)->getAllProducts()->selectRaw('count(*) as prdct_count, category, categories.title, categories.slug, categories.parent_id')->leftJoin('categories', 'products.category', '=', 'categories.id')->where('products.status', 'publish')->groupBy('products.category')->get();
+            $mainCategories = Category::where('parent_id', 0)->pluck('slug','id');
             foreach ($productCategories as $productCategory) {
                 if ($productCategory->category != 0) {
-                    $categoryDetails = Category::find($productCategory->category);
-                    $mainCategoryDetails = Category::where('id', $categoryDetails->parent_id)->where('parent_id', 0)->first();
-                    $productSubCategories = Product::select(DB::raw("count(*) as prdct_count"), "sub_category")->where('status', 'publish')->where('category', $productCategory->category)->where('user_id', $userId)->groupBy('sub_category')->get();
+
+                    $productSubCategories = User::find($userId)->getAllProducts()->selectRaw("count(*) as prdct_count, sub_category, categories.title, categories.slug")->leftJoin('categories', 'products.sub_category', '=', 'categories.id')->where('products.status', 'publish')->where('category', $productCategory->category)->groupBy('sub_category')->get();
```

Comment:
Instead of querying like User::find($userId)->getAllProducts()...
do it like $productCategories->selectRaw..................
you'll get the $productCategories when you will make the changes from above comment.